### PR TITLE
First pass reverts, fixes, and contribs for 1.14

### PR DESCRIPTION
**Pull request by ProgrammerDan**
Reverting some null checks, I know this will cause warnings in IDE but some were hard earned. I do hope that Spigot is more faithful and careful in these later releases, but better to check and be sure then not check and be unsure. Some I did not add back in. Any I added back in I wrote why I added it. I also switched around some orderings to use equals() checks, and also addressed some changes to map views in new releases, although there is more to do there. As well, expanded certain block lists and biome lists even further, lots of new content in these new minecraft releases. Probably first of several commits, as I test.

I'll unlock from draft once I'm confident this is working.

### CODE REVIEW

File: src/main/java/com/programmerdan/minecraft/simpleadminhacks/hacks/Experimental.java [hunk 1]
```diff
@@ -153,8 +163,16 @@ private void logTeleport(PlayerTeleportEvent event, StringBuilder sb) {
 		Player player = event.getPlayer();
 		Location from = event.getFrom();
 		Location to = event.getTo();
-		sb.append(String.format(" %16s", player.getName()));
-		sb.append(String.format(" %s,%5.0f,%3.0f,%5.0f", from.getWorld().getName(), from.getX(), from.getY(), from.getZ()));
+		// Must of these nullchecks will gen warnings with compile time contracts.
```
Comment (ProgrammerDan) [inline src/main/java/com/programmerdan/minecraft/simpleadminhacks/hacks/Experimental.java]: Must --- Most

File: src/main/java/com/programmerdan/minecraft/simpleadminhacks/hacks/GameTuning.java [hunk 2]
```diff
@@ -325,16 +326,16 @@ public void onPlayerInteractEntity(PlayerInteractEntityEvent event) {
 	@EventHandler(priority = EventPriority.LOW, ignoreCancelled = true)
 	public void onEntityChangeBLock(EntityChangeBlockEvent event) {
 		if(config.isEnabled() &&
-				(!config.isEnderGrief() && event.getEntityType() == EntityType.ENDERMAN) ||
-				(!config.isWitherGrief() && event.getEntityType() == EntityType.WITHER)) {
+				(!config.isEnderGrief() && EntityType.ENDERMAN.equals(event.getEntityType())) ||
+				(!config.isWitherGrief() && EntityType.WITHER.equals(event.getEntityType()))) {
```
Comment (ProgrammerDan) [inline src/main/java/com/programmerdan/minecraft/simpleadminhacks/hacks/GameTuning.java]: What method does a dragon use? Any new mobs cause change events? Will look.

File: src/main/java/com/programmerdan/minecraft/simpleadminhacks/hacks/TimingsHack.java [hunk 3]
```diff
@@ -256,9 +257,10 @@ public boolean onCommand(CommandSender sender, Command command, String label, St
 				view.getRenderers().forEach(view::removeRenderer);
 				view.addRenderer(this.tickVisualize);
 
-				ItemStack viewMap = new ItemStack(Material.MAP, 1, view.getId());
+				ItemStack viewMap = new ItemStack(Material.MAP, 1);
```
Comment (ProgrammerDan) [inline src/main/java/com/programmerdan/minecraft/simpleadminhacks/hacks/TimingsHack.java]: I need to review this whole structure here. Might be a better way now, with API rewrites.

File: README.md [hunk 4]
```diff
@@ -107,6 +109,23 @@ are actively breaking reinforced containers.
...
 all breaks or builds under a specific y level.
 
+### ToggleLamp
+
+A solid piece of contributed code, this hack lets redstone lamps be used as
+clickable-on, clickable-off light sources without additional redstone infrastructure.
+
+### HumbugBatchOne
+
+A new BasicHack (thanks Maxopoly) which encompases a number of previously unincluded
```
Comment (ProgrammerDan) [inline README.md]: encompases --> encompasses

File: src/main/java/com/programmerdan/minecraft/simpleadminhacks/hacks/GameTuning.java (hunk 2 above)
Comment (ProgrammerDan) [reply src/main/java/com/programmerdan/minecraft/simpleadminhacks/hacks/GameTuning.java]: no new mobs that I can see, but dragon easily added and withers can still destroy with wither skulls, so added that. Might need to gate wither skull damage and dragon fireball damage separately...

Comment (ProgrammerDan) [reply src/main/java/com/programmerdan/minecraft/simpleadminhacks/hacks/GameTuning.java]: Added dragon

**Comment by Maxopoly** [comment]
Is this still being worked on? Github will not let me merge it as long as it's a draft.

**Comment by ProgrammerDan** [comment]
Yes, apologies. Been in a funk

**Comment by Maxopoly** [comment]
Note that I fixed some other stuff myself, so pull before you make further changes

**Comment by ProgrammerDan** [comment]
Thanks, will do

File: src/main/java/com/programmerdan/minecraft/simpleadminhacks/hacks/GameTuning.java [hunk 5]
```diff
@@ -210,7 +211,7 @@ public void enterBed(PlayerBedEnterEvent e) {
 		if (!config.isEnabled() || !config.areDaytimeBedsEnabled()) {
 			return;
 		}
-		if (e.getBedEnterResult() == BedEnterResult.NOT_POSSIBLE_NOW || e.getBedEnterResult() == BedEnterResult.NOT_SAFE) {
+		if (BedEnterResult.NOT_POSSIBLE_NOW.equals(e.getBedEnterResult()) || BedEnterResult.NOT_SAFE.equals(e.getBedEnterResult())) {
```
Comment (Maxopoly) [inline src/main/java/com/programmerdan/minecraft/simpleadminhacks/hacks/GameTuning.java]: I noticed you did this in many other places as well and actually changed it the other way around in some of my commits. Why use equals() on Enums? == is faster and easier to read.

**Comment by ProgrammerDan** [comment]
standard form.

Avoids any problem if it's not actually an enum (happens in several places
in Spigot) and in general is a safer form. Yes not required for enums
specifically but I don't prefer to switch between forms and allows for
visual conformity.

It was the standard we were moving to in our last rewrites so I've just
kept doing it. I understand if you disagree, but I much prefer the single
standard instead of switching dependent on context.

**Comment by Maxopoly** [comment]
I merged your PR and my continued work into the new branch civ14_2, because SAH was the last plugin missing on my update journey. Currently in testing phase with all the plugins https://www.reddit.com/r/civclassics/comments/dn5vk5/114/

**Comment by ProgrammerDan** [comment]
Sorry for holding you up, amazing work.

**Pull request merged**